compiler-testing/src/main/kotlin/com/squareup/anvil/compiler/testing/compilation/Compile2.kt

```diff
@@ -0,0 +1,471 @@
+package com.squareup.anvil.compiler.testing.compilation
+
+import com.rickbusarow.kase.stdlib.createSafely
+import com.rickbusarow.kase.stdlib.div
+import com.rickbusarow.kase.stdlib.letIf
+import com.squareup.anvil.compiler.k2.fir.AnvilFirExtensionFactory
+import com.squareup.anvil.compiler.testing.CompilationEnvironment
+import com.squareup.anvil.compiler.testing.kotlinVersion
+import dagger.internal.codegen.ComponentProcessor
+import io.kotest.matchers.shouldBe
+import org.intellij.lang.annotations.Language
+import org.jetbrains.kotlin.cli.common.ExitCode
+import org.jetbrains.kotlin.cli.common.arguments.K2JVMCompilerArguments
+import org.jetbrains.kotlin.cli.common.arguments.copyOf
+import org.jetbrains.kotlin.cli.common.arguments.parseCommandLineArguments
+import org.jetbrains.kotlin.cli.jvm.K2JVMCompiler
+import org.jetbrains.kotlin.compilerRunner.toArgumentStrings
+import org.jetbrains.kotlin.config.LanguageVersion
+import org.jetbrains.kotlin.incremental.isJavaFile
+import org.jetbrains.kotlin.kapt3.base.AptMode
+import org.jetbrains.kotlin.kapt3.base.DetectMemoryLeaksMode
+import org.jetbrains.kotlin.kapt3.base.KaptFlag
+import org.jetbrains.kotlin.kapt3.base.KaptOptions
+import java.io.File
+import java.io.InputStream
+import java.io.OutputStreamWriter
+import java.io.PrintStream
+import java.net.URI
+import java.net.URLClassLoader
+import javax.tools.Diagnostic
+import javax.tools.DiagnosticCollector
+import javax.tools.JavaFileObject
+import javax.tools.SimpleJavaFileObject
+import javax.tools.ToolProvider
+
+public fun CompilationEnvironment.compile2(
```

Comment:
I think this would be a good place to document why we have compile2 and any issues encountered with KCT / how compile2 resolves them, especially for folks who may run into those issues in the future when updating their custom code generator tests